internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource.go

```diff
@@ -274,14 +281,22 @@ func resourceLogAnalyticsClusterCustomerManagedKeyDelete(d *pluginsdk.ResourceDa
 		return fmt.Errorf("deleting `azurerm_log_analytics_cluster_customer_managed_key` %s: `customer managed key does not exist!`", *id)
 	}
 
+	// Ensure `associatedWorkspaces` is not present in request, this is a read only property and cannot be sent to the API
+	// Error: updating Customer Managed Key for Cluster
+	//		performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error:
+	//		InvalidParameter: 'properties.associatedWorkspaces' is a read only property and cannot be set.
+	//		Please refer to https://docs.microsoft.com/en-us/azure/azure-monitor/log-query/logs-dedicated-clusters#link-a-workspace-to-the-cluster for more information on how to associate a workspace to the cluster.
+	props.AssociatedWorkspaces = nil
+
+	// The API only removes the CMK when it is sent empty string values, sending nil for each property or an empty object does not work.
 	model.Properties.KeyVaultProperties = &clusters.KeyVaultProperties{
-		KeyVaultUri: nil,
-		KeyName:     nil,
-		KeyVersion:  nil,
+		KeyVaultUri: pointer.To(""),
+		KeyName:     pointer.To(""),
+		KeyVersion:  pointer.To(""),
 	}
 
 	if err = client.CreateOrUpdateThenPoll(ctx, *id, *model); err != nil {
-		return fmt.Errorf("updating Customer Managed Key for %s: %+v", *id, err)
+		return fmt.Errorf("removing Customer Managed Key from %s: %+v", *id, err)
```

Comment:
Since it's the delete does it make sense for this to be 
```suggestion
		return fmt.Errorf("deleting Customer Managed Key from %s: %+v", *id, err)
```